db/migrate/20160621151100_polymorphic_mentions.rb

```diff
@@ -0,0 +1,29 @@
+class PolymorphicMentions < ActiveRecord::Migration
+  def change
+    remove_index :mentions, column: %i(post_id)
+    remove_index :mentions, column: %i(person_id post_id), unique: true
+    rename_column :mentions, :post_id, :mentions_container_id
+    add_column :mentions, :mentions_container_type, :string, null: false
+    add_index :mentions,
+              %i(mentions_container_id mentions_container_type),
+              name:   "index_mentions_on_mc_id_and_mc_type",
+              length: {mentions_container_type: 191}
+    add_index :mentions,
+              %i(person_id mentions_container_id mentions_container_type),
+              name:   "index_mentions_on_person_and_mc_id_and_mc_type",
+              length: {mentions_container_type: 191}
+
+    reversible(&method(:up_down))
+  end
+
+  def up_down(change)
+    change.up do
+      Mention.update_all(mentions_container_type: "Post")
```

Comment:
Can you add [migration models](http://blog.makandra.com/2010/03/how-to-use-models-in-your-migrations-without-killing-kittens/) for the migrations?